stable_baselines/acer/acer_simple.py

```diff
@@ -180,7 +180,7 @@ def setup_model(self):
                 self.sess = tf_util.make_session(num_cpu=self.num_procs, graph=self.graph)
 
                 n_batch_step = None
-                if issubclass(self.policy, LstmPolicy):
+                if issubclass(self.policy, StatefulActorCriticPolicy):
```

Comment:
Maybe rather `assert issubclass(self.policy, ActorCriticPolicy)` somewhere before and then just use the `stateful` flag? Same for a2c, actr_disc and PPO2

Reply:
These RL algorithms depend on `masks_ph` and `states_ph` which are defined `StatefulActorCriticPolicy` but not guaranteed to exist in a policy that is just `stateful`. This change might make sense after addressing some of the points in the previous comment though so leaving open for now.

Reply:
Yep, in some way we need to expose masks_ph and states_ph from ANY staful policy so the subclass test here is not needed any more. I agree that the issues above need to be fixed first.